### PR TITLE
Fix feedforward external features

### PR DESCRIPTION
**Pull request by StanislasGuinel**
This PR must fix feedforward with external features and multiple timeseries of different lengths

### CODE REVIEW

File: python-lib/gluonts_forecasts/model_handler.py [hunk 1]
```diff
@@ -223,3 +223,11 @@ def get_model_label(model_name):
         return None
     else:
         return model_descriptor.get(LABEL, "")
+
+
+def get_model_name_from_label(model_label):
```
Comment (alexcombessie) [inline python-lib/gluonts_forecasts/model_handler.py:228]: Perhaps there is a simpler way to do this?

File: python-lib/gluonts_forecasts/trained_model.py [hunk 2]
```diff
@@ -25,7 +28,8 @@ class TrainedModel:
         forecasts_df (DataFrame): Dataframe with the different quantiles forecasts and the training data if include_history is True
     """
 
-    def __init__(self, predictor, gluon_dataset, prediction_length, quantiles, include_history, history_length_limit=None):
+    def __init__(self, model_name, predictor, gluon_dataset, prediction_length, quantiles, include_history, history_length_limit=None):
```
Comment (alexcombessie) [inline python-lib/gluonts_forecasts/trained_model.py:31]: Can't the model_name be derived from the `predictor`? Do we need to add it as a mandatory attribute?

File: tests/python/unit/test_model.py [hunk 3]
```diff
@@ -121,3 +123,77 @@ def metrics_assertions(metrics, model_name):
     def forecasts_assertions(forecasts_df, model_name, prediction_length=1):
         assert len(forecasts_df.index) == 2
         assert forecasts_df["index"].nunique() == prediction_length
+
+
+class TestExternalFeaturesModel:
```
Comment (alexcombessie) [inline tests/python/unit/test_model.py:128]: Suggestion: Specify SimpleFeedForward in the test class name?

File: python-lib/gluonts_forecasts/trained_model.py (hunk 2 above)
Comment (StanislasGuinel) [reply python-lib/gluonts_forecasts/trained_model.py:31]: all the deep learning models predictor are of the same class (gluonts.mx.model.predictor.RepresentableBlockPredictor)
I can make it optional and use the previous behavior (don't remove anything from the list_dataset) when no name is set

File: python-lib/gluonts_forecasts/model_handler.py (hunk 1 above)
Comment (StanislasGuinel) [reply python-lib/gluonts_forecasts/model_handler.py:228]: sure, I will do a one-liner :)

File: python-lib/gluonts_forecasts/trained_model.py (hunk 2 above)
Comment (alexcombessie) [reply python-lib/gluonts_forecasts/trained_model.py:31]: Okay, let's keep it then, good idea. Closing...

File: tests/python/unit/test_model.py (hunk 3 above)
Comment (StanislasGuinel) [reply tests/python/unit/test_model.py]: done in 
d42615b

File: python-lib/gluonts_forecasts/trained_model.py (hunk 2 above)
Comment (StanislasGuinel) [reply python-lib/gluonts_forecasts/trained_model.py]: done in d42615b

File: python-lib/gluonts_forecasts/model_handler.py (hunk 1 above)
Comment (StanislasGuinel) [reply python-lib/gluonts_forecasts/model_handler.py:228]: done in d42615b

Comment (alexcombessie) [reply python-lib/gluonts_forecasts/model_handler.py:228]: Nice!

**Review by alexcombessie — APPROVED** [review]

**Pull request merged**